intelmq/lib/message.py

```diff
@@ -304,17 +304,32 @@ def __get_type_config(self, key):
     def __hash__(self):
         return int(self.hash(), 16)
 
-    def hash(self, blacklist=frozenset()):
+    def hash(self, filter_type="blacklist", filter_keys_list=frozenset()):
         """Return a sha256 hash of the message as a hexadecimal string.
-        The hash is computed over almost all key/value pairs. The only
-        keys omitted are 'time.observation' and all keys contained in
-        the optional blacklist parameter. If given, the blacklist
+        The hash is computed over almost all key/value pairs. Depending on
+        filter_type parameter (blacklist or whitelist), the keys defined in
+        filter_keys_list parameter will be considered as the keys to ignore
+        or the only ones to consider. If given, the filter_keys_list
         parameter should be a set.
         """
+
+        if filter_type not in ["whitelist", "blacklist"]:
+
+            raise exceptions.InvalidArgument('filter_type',
+                                             got=filter_type,
+                                             expected='<whitelist> or'
```

Comment:
Should be a list of values: `['whitelist', 'blacklist']`, the formatting is done by InvalidArgument anyway.